### PR TITLE
feat(scripts): adding hdwallet to switch between ganache and Ropsten testnet

### PR DESCRIPTION
**Pull request by jeff1010322**
### Description of the Change

* Replaced the web3.js with hdwallet.js to generate the accounts using a Truffle HDWallet Provider.
  * To use with local Ganache must now at least set the MNEMONIC environment variable with the ganache generated mnemonic.
  * To use with Ropsten testnet set the MNEMONIC and the NODE_ADDR environment variables. 

* Switched environment variable CONTRACT_ADDRESS to AUCTION_ADDRESS

* Added scripts pause.js and whitelist.js to help with Ropsten testing.  
  * pause.js takes 2 parameters: pause or unpause followed by token or auction to allow easily pausing or unpausing either of the token contracts or auction contracts.
  * whitelist.js takes an address as the only parameter to whitelist an account for the auction contract.  Account[0] generated by the HDWallet provider must be the owner of the auction contract.

### CODE REVIEW

**Comment by xjerod** [comment]
Might be worth noting somewhere that these shouldn't be used on main net as-is due to the way we're setting the gas for the transactions.

**Comment by elewis787** [comment]
Agree with Jerod. Other than that lgtm

File: scripts/whitelist.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/* eslint no-unused-vars: "off", prefer-const: "off" */
+const {
+  getAccounts, whitelist,
+} = require('./hdwallet');
+
+async function addToWhitelist(address) {
+    const accounts = await getAccounts();
+
+    console.log('Whitelisting address', address);
+    await whitelist(myArgs[0], address);
```
Comment (xjerod) [inline scripts/whitelist.js]: This line should be `await whitelist(address, accounts[0]);` address = user input = myArgs[0] so this is passing the same address for both params, and the params for the whitelist function is `(addressToBeWhitelisted, owner)`

**Review by xjerod — CHANGES_REQUESTED** [review]

File: scripts/whitelist.js (hunk 1 above)
Comment (jeff1010322) [reply scripts/whitelist.js]: Good catch, I fixed the parameters in the latest commit.

Comment (xjerod) [reply scripts/whitelist.js]: LGTM

**Review by xjerod — APPROVED** [review]

**Pull request merged**